### ISSUE TITLE
Compatibility mode project windows jump up after they appear on screen

### ISSUE THREAD

**Issue by okla** [godotengine/godot#79581]
### Godot version

v4.1.stable.official [970459615]

### System information

Godot v4.1.stable - Fedora Linux 38 (Cinnamon) - Vulkan (Compatibility) - Mesa Intel(R) UHD Graphics 630 (CML GT2) () - Intel(R) Core(TM) i5-10505 CPU @ 3.20GHz (12 Threads)

### Issue description

Compatibility mode project's main window and all windows created and shown from code jump up after they appear on screen. There was no such issue in 4.0.3. There is also no such issue in Forward+/Mobile mode.

### Steps to reproduce

1. Create a new project in Compatibility mode
2. Create main scene with any node in it
3. Run the project

### Minimal reproduction project

[Bug.zip](https://github.com/godotengine/godot/files/12069068/Bug.zip)

**Comment by clayjohn** [comment]
What do you mean by "jump up"?

**Comment by okla** [comment]
@clayjohn Here are the subsequent frames of launching the attached project recorded in 4.1.1. As you can see in the last frame the window moves up (to the height of its top decoration I guess). There was no such behavior before 4.1.

![1](https://github.com/godotengine/godot/assets/8558017/371e6ffb-615b-464f-a2e5-3f61b7e7302c)
![2](https://github.com/godotengine/godot/assets/8558017/95fbeb55-828f-40bd-b2f1-d652cbd38c3d)
![3](https://github.com/godotengine/godot/assets/8558017/d9dae2b8-55b1-4143-8fd2-affb71bdd757)

**Comment by okla** [comment]
@Calinou It's X11. I'll test it on the betas a little later.

**Comment by okla** [comment]
@Calinou I've tested this issue on previous versions and it exists since 4.1.dev1 (4.2.dev2 is affected too). It's also not exclusive to compatibility mode and appears in all rendering modes (will correct the issue text and title).

**Comment by MightiestGoat** [comment]
Is this occurring with --single-window?

**Comment by okla** [comment]
@MightiestGoat Yeah, this occurs with `--single-window` (if I understand correctly that it should be passed to the editor on launch).

Unfortunately, I don't have another WM to test.

**Comment by reptofrog** [comment]
Same thing happens on GNOME Xorg and GNOME Wayland. Here's a video:
[screencast.webm](https://github.com/godotengine/godot/assets/36744029/162e6981-408a-497b-ae42-e2d3b2307f72)

**Comment by okla** [comment]
What else can be done to fix this issue? Since it's a regression in 4.1.dev1 maybe some commit between it and 4.0.3 can simply be reverted?

**Comment by akien-mga** [comment]
@okla You can try to revert #76040 locally and see if that's indeed what caused the change.

As you can see in that PR and the linked issue, it fixes an issue which is the opposite of what you're experiencing, so simply reverting it won't really improve things, it will just make it worse for others.
